CONTRIBUTING.md

```diff
@@ -344,52 +344,43 @@ Consumers that violate the spec by throwing exceptions for parsing errors at run
 
 ## Writing Asynchronous Tests
 
-An asynchronous test is any test that include the `async` frontmatter flag. When executing such tests, the runner expects that the global `$DONE()` function will be called **exactly once** to signal test completion.
+An asynchronous test is any test that include the `async` frontmatter flag.
 
- * If the argument to `$DONE` is omitted, is `undefined`, or is any other falsy value, the test is considered to have passed.
-
- * If the argument to `$DONE` is a truthy value, the test is considered to have failed and the argument is displayed as the failure reason.
-
-A common idiom when writing asynchronous tests is the following:
+For most asynchronous tests, the `asyncHelpers.js` harness file includes an `asyncTest` method that precludes needing to interact with the test runner via the `$DONE` function. `asyncTest` takes an async function and will ensure that `$DONE` is called properly if the async function returns or throws an exception.` For example, a test written using `asyncTest` might look like:
 
 ```js
-var p = new Promise(function () { /* some test code */ });
-
-p.then(function checkAssertions(arg) {
-  if (!expected_condition) {
-    throw new Test262Error("failure message");
-  }
+/*---
+... (other frontmatter) ...
+flags: [async]
+includes: [asyncHelpers.js]
+---*/
 
-}).then($DONE, $DONE);
+asyncTest(async function() {
+  assert.sameValue(true, await someTestCode(1), "someTestCode(1) should return true");
+;
+});
 ```
 
-Function `checkAssertions` implicitly returns `undefined` if the expected condition is observed.  The return value of function `checkAssertions` is then used to asynchronously invoke the first function of the final `then` call, resulting in a call to `$DONE(undefined)`, which signals a passing test.
+For more complicated asynchronous testing, such as testing Promise or other core asynchronous functionality, the runner expects that the global `$DONE()` function will be called **exactly once** to signal test completion.
 
-If the expected condition is not observed, function `checkAssertions` throws a `Test262Error`.  This is caught by the Promise and then used to asynchronously invoke the second function in the call -- which is also `$DONE` -- resulting in a call to `$DONE(error_object)`, which signals a failing test.
+ * If the argument to `$DONE` is omitted, is `undefined`, or is any other falsy value, the test is considered to have passed.
+
+ * If the argument to `$DONE` is a truthy value, the test is considered to have failed and the argument is displayed as the failure reason.
 
 ### Checking Exception Type and Message in Asynchronous Tests
```

Comment:
```suggestion
### Checking Exception Type in Asynchronous Tests
```
We no longer have an example of how to check the exception message, but also I think we don't need one because I don't think any tests I've seen do so. Nor should they, since the contents of exception messages are not part of the requirements for a JS implementation.